Add support for a Composer script to install theme assets

mbabker opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | no|
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | N/A |
| License         | MIT |

Symfony (through the `SensioDistributionBundle`) has a hook point to trigger the `assets:install` command after a `composer install` or `composer update`.  When using the `SyliusThemeBundle`, you basically need to run `sylius:theme:assets:install` at the same time to handle having assets installed in the right place.  To simplify this, we can also create a Composer hook to do the job for us.

This PR adds a new class extending `Sensio\Bundle\DistributionBundle\Composer\ScriptHandler` to add support for triggering the `sylius:theme:assets:install` command.  To use it, one would just need to add `"Sylius\\Bundle\\ThemeBundle\\Composer\\ScriptHandler::installThemeAssets"` to their `post-install-cmd` and `post-update-cmd` scripts.

stefandoorn commented:
> To use it, one would just need to add "Sylius\\Bundle\\ThemeBundle\\Composer\\ScriptHandler::installThemeAssets" to their post-install-cmd and post-update-cmd scripts.

Wouldn't it be nicer to add it by default to `composer.json` and inside the script check whether there is anything to do? Maybe the command already doesn't do anything if there are no themes? Else it should probably be mentioned in the docs, but I'm afraid many people will overlook such a handy feature then..

mbabker commented:
I literally got the idea 10 minutes ago so just copied the base stuff to get started.  We can definitely do more with it (docs, put it in `composer.json`, whatever), just wanted to get the ball rolling first 😄

stefandoorn commented:
Hehe ok :) I find it a very handy feature anyway :)

mbabker commented:
This should be good for review now.

I did a couple of tests against this repo and a project where I am using themes, in both cases `sylius:theme:assets:install` handles the default `assets:install` command's behavior just fine as well as adding the resources for the theme.  So I've changed the `composer.json` to use this command in place of the default `"Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets"`.

I've also updated the bundle docs a bit to add the Composer script handler and note that the command can basically replace `assets:install`.

mbabker commented:
Unfortunately even `SensioDistributionBundle` doesn't have any tests, so there's not even a boilerplate to go off of like there was for the production code.  So that'll be a "fun" challenge.
